jujugui/static/gui/src/app/store/env/base.js

```diff
@@ -366,32 +367,46 @@ YUI.add('juju-env-base', function(Y) {
      * Retrieve the stored user credentials.
      *
      * @method getCredentials
-     * @return {Object} The stored user credentials with a 'user' and a
-     *                   'password' attribute.
+     * @return {Object} The stored user credentials with "user", "password" and
+     *   "macaroons" attributes. This object also exposes the "areAvailable"
+     *   property holding whether either user/password or macaroons credentials
+     *   are actually available.
      */
     getCredentials: function() {
       var credentials = JSON.parse(
-          module.sessionStorage.getItem('credentials'));
-      if (credentials) {
-        // All juju-core interactions require the 'user-' prefix on the username
-        // but we hide that from the user so we prefix that here instead.
-        if (credentials.user && credentials.user.indexOf(API_USER_TAG) !== 0) {
+        module.sessionStorage.getItem('credentials'));
+      if (!credentials) {
+        credentials = {};
```

Comment:
You can disregard all of my comments about this method not returning an object :D